src/Service/Features/FeatureContextProvider.cs

```diff
@@ -47,7 +47,7 @@ public class FeatureContextProvider : IFeatureContextProvider
 
             if (string.IsNullOrWhiteSpace(appId)) return new NullFeaturesContext();
 
-            var storage = _storageFactory.Create(appId);
+            var storage = _tenantStorageFactory.Create(appId);
```

Comment:
question: Does this run after Authz? Is that why we can't inject ITenantStorage directly?

Reply:
This can even run before authentication. It's used by event logging. Could even be triggered when logging bad api keys for example. When that happens, the claims will not be populated.

Some endpoints in the admin console also call the get features endpoint which leverages the FeaturesContextProvider. The reason that was done there was that the FeaturesContextProvider could have already been instantiated in the middleware, so the `FeaturesContext` is already in memory. So it was much quicker to access the `FeaturesContextProvider`.